File: assignments/scrabble_score/javascript/scrabble_score_test_runner.js
const assert = require('assert');
const { ScrabbleScore } = require('./scrablle_score.js'); // Assuming student file

function runTests() {
    const solver = new ScrabbleScore();
    // Test Cases: { word1: string|null, word2: string|null, expected: string }
    const testCases = [
        { word1: "Question?", word2: "Query", expected: "One wins!" },
        { word1: "Apple", word2: "Banana", expected: "Two wins!" },
        { word1: "Programming", word2: "Coding", expected: "One wins!" },
        { word1: "HELLO", word2: "hello", expected: "Tie!" },
        { word1: "CS50", word2: "Harvard", expected: "Two wins!" },
        { word1: "", word2: "Empty", expected: "Two wins!" },
        { word1: "Points!", word2: "Score?", expected: "One wins!" },
        { word1: "Tie", word2: "Tie", expected: "Tie!" },
        { word1: "Test", word2: "", expected: "One wins!" },
        { word1: null, word2: "Word", expected: "Two wins!" },
        { word1: "Word", word2: null, expected: "One wins!" },
        { word1: null, word2: null, expected: "Tie!" }
    ];

    let passedCount = 0;
    let report = [];
    console.log("===== Starting Automated Tests for Scrabble Score Comparator (JavaScript) =====");

    testCases.forEach((tc, i) => {
        let actual = "Error";
        const input1Desc = tc.word1 === null ? "null" : `"${tc.word1}"`;
        const input2Desc = tc.word2 === null ? "null" : `"${tc.word2}"`;
        const caseIdentifier = `Test Case #${i + 1}: Player 1(${input1Desc}) vs Player 2(${input2Desc})`;
        try {
            // Call the updated method
            actual = solver.compareScores(tc.word1, tc.word2);
            assert.strictEqual(actual, tc.expected);
            report.push(`${caseIdentifier} -> ✅ PASS (Result: "${actual}")`);
            passedCount++;
        } catch (error) {
             report.push(`${caseIdentifier} -> ❌ FAIL/ERROR: Expected "${tc.expected}", but got "${actual}"`);
        }
        report.push(''); // Add blank line
    });

    console.log(report.join('\n'));
    console.log("===== Test Summary =====");
    console.log(`Total Tests: ${testCases.length}`); console.log(`Passed: ${passedCount}`); console.log(`Failed: ${testCases.length - passedCount}`); console.log("========================");
    console.log(`[SUMMARY]:${passedCount}:${testCases.length}`);
}

runTests();

File: assignments/scrabble_score/python/scrabble_score_test_runner.py
import sys
from scrabble_score import ScrabbleScore # Import the student's class

def run_tests():
    solver = ScrabbleScore()
    # Test Cases: (word1, word2, expectedResultString)
    test_cases = [
        ("Question?", "Query", "One wins!"), ("Apple", "Banana", "Two wins!"),
        ("Programming", "Coding", "One wins!"), ("HELLO", "hello", "Tie!"),
        ("CS50", "Harvard", "Two wins!"), ("", "Empty", "Two wins!"),
        ("Points!", "Score?", "One wins!"), ("Tie", "Tie", "Tie!"),
        ("Test", "", "One wins!"), (None, "Word", "Two wins!"), # Test None
        ("Word", None, "One wins!"), (None, None, "Tie!")
    ]
    passed_count = 0
    report = []
    print("===== Starting Automated Tests for Scrabble Score Comparator (Python) =====")

    for i, (word1, word2, expected) in enumerate(test_cases):
        actual = "Error"
        input1_desc = "None" if word1 is None else f'"{word1}"'
        input2_desc = "None" if word2 is None else f'"{word2}"'
        case_identifier = f'Test Case #{i + 1}: Player 1({input1_desc}) vs Player 2({input2_desc})'
        try:
            # Call the updated method
            actual = solver.compare_scores(word1, word2)
            report.append(case_identifier)
            report.append(f' -> Expected: "{expected}", Got: "{actual}"')

            if actual == expected:
                report.append(" -> ✅ PASS\n")
                passed_count += 1
            else:
                report.append(" -> ❌ FAIL\n")
        except Exception as e:
            report.append(case_identifier)
            report.append(f" -> ❌ ERROR during comparison: {type(e).__name__}\n")

    print("\n".join(report))
    print("===== Test Summary ====="); print(f"Total Tests: {len(test_cases)}"); print(f"Passed: {passed_count}"); print(f"Failed: {len(test_cases) - passed_count}"); print("========================")
    print(f"[SUMMARY]:{passed_count}:{len(test_cases)}")

if __name__ == "__main__":
    sys.path.append('.')
    run_tests()